Results_Engine/Query/ResultItemValueProperty.cs

```diff
@@ -0,0 +1,132 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using BH.oM.Analytical.Results;
+using BH.oM.Quantities.Attributes;
+using System;
+using System.Collections.Generic;
+using System.Collections;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using BH.Engine.Base;
+
+namespace BH.Engine.Results
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns a Fuction for extracting a value from a result item.")]
+        [Input("result", "The result used to extract the type to generate the Func for value extraction.")]
+        [Input("propertyName", "The name of the property to get the function for.")]
+        [MultiOutput(0, "propertyName", "The name of the property or method function.")]
+        [MultiOutput(1, "func", "The result extraction function.")]
+        [MultiOutput(2, "quantity", "The quantity corresponding to the function and property.")]
+        public static Output<string, Func<IResultItem, double>, QuantityAttribute> ResultItemValueProperty(this IResultItem result, string propertyName)
```

Comment:
Maybe worth formalising the Output into a class, to make it easier to reuse this function and avoiding Tuples?

We could have a class `ResultItemValueProperty` with those three properties (PropertyName, ExtractorFunction, Quantity)